### PR TITLE
Moved symbolization code shared by S3 input and SQS input plugins

### PR DESCRIPTION
**Pull request by andsel**
## Release notes

[rn:skip]

## What does this PR do?
Moves symbolization code of `additional_settings` used by S3 input and SQS input into common place.
If the including plugin defines the config attribute `addistiona_settings` then the method `aws_options_hash` symbolize the `additional_settings` map, merging the outcome into the options available to the plugin itself. 

## Why is it important/What is the impact to the user?
As a developer it reduces the need to copy and paste code to provide `additional_settings` config.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files (and/or docker env variables)~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~

## Author's Checklist

- [x] build locally, install on a LS instance and check the the existing SQS input (the same is valid for S3 input) doesn't break when including this mixin.

## How to test this PR locally
- build the gem `gem build`
- install on a local LS instance: `bin/ruby -S gem install /hpath_to/logstash-mixin-aws/logstash-mixin-aws-5.1.0.gem`
- remove (if present) the SQS input and reinstall so that the `Gemfile.lock` takes the update to 
```
logstash-mixin-aws (5.1.0)
```
- run a LS pipeline with SQS input and check it still works.

## Related issues

-

### CODE REVIEW

File: lib/logstash/plugin_mixins/aws_config/v2.rb
Comment (kares) [inline lib/logstash/plugin_mixins/aws_config/v2.rb:35]: should maybe merge the [other way around for compatibility](https://github.com/logstash-plugins/logstash-input-s3/blob/v3.8.3/lib/logstash/inputs/s3.rb#L444) as well as not to override explicit settings:
```
  endpoint => 'foo'
  additional_settings => { endpoint => 'bar' }
```
... would end up with `endpoint => bar`, while the current behavior (in S3) is that it ends up as `endpoint => foo`

**Review by kares — COMMENTED** [review]
looking on track - one suggestion in terms of `additional_settings` overriding explicit configuration, 

also think despite having the `additional_settings` handling in the plugins for a while, until they get updated, in terms of functionality it should work even if the settings are set "twice" ...

File: lib/logstash/plugin_mixins/aws_config/v2.rb
Comment (andsel) [reply lib/logstash/plugin_mixins/aws_config/v2.rb]: Good point, fixed with [190bfa5](https://github.com/logstash-plugins/logstash-mixin-aws/pull/53/commits/190bfa5bbafb01516d975fa0de7ae812001ef3d7)

Comment (andsel) [reply lib/logstash/plugin_mixins/aws_config/v2.rb]: @kares if you want it's ready for a second review

**Review by kares — APPROVED** [review]
LGTM

**Pull request merged**
